common/types/data_connections.ts

```diff
@@ -83,19 +83,19 @@ export interface CachedColumn {
 
 export interface CachedTable {
   name: string;
-  columns: CachedColumn[];
+  columns?: CachedColumn[];
```

Comment:
Why optional here?

Reply:
Cause columns are lazy loaded for tables. These will be loaded in cache when users open `table flyout` or use a table in `create acceleration flyout`.